hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -361,7 +366,15 @@ private static URI getS3Endpoint(String endpoint, final Configuration conf) {
    */
   private static Region getS3RegionFromEndpoint(String endpoint) {
```

Comment:
this could be made package private/visible for testing and then have some unit tests which will run every time yetus does its work. this should include a test which passes an endpoint which shouldn't match the regexp -and verifies that it is rejected

Reply:
Added some unit tests on endpoint parsing.